docker/java-and-python/build.gradle

```diff
@@ -0,0 +1,13 @@
+plugins {
+    id 'com.bmuschko.docker-remote-api'
+}
+
+evaluationDependsOn Docker.registryProject('debian')
+
+def buildDocker = Docker.registerDockerImage(project, 'buildDocker') {
+    inputDir.set layout.projectDirectory.dir('src/main/docker')
+    inputs.files Docker.registryFiles(project, 'debian')
```

Comment:
if we're going so far as to providing all our own tagged uptsreams, why not take the next step and provide a model for these, let registerDockerImage have a reference to the string in the registry? then it can invoke evaluationDependsOn and find the files itself?

Reply:
This seems like an ok idea - it's just a deeper level of integration that I didn't want to go down at this time. Giving the PR a "lite" touch.

Reply:
To verify, do we need to pin any of the entries currently present? It looks like after you run `pullImage` (without cleaning docker...) you will be reminded each time about alpine if you run `bumpImage`, with no way to suppress it?

I know we're accidentally out of date on this branch and so expecting the command to fail for most of them, but the new github actions check added in this commit will fail on the deliberately-out-of-date alpine, right?

Reply:
I *think* you mean `compareImage` and not `bumpImage`? You'll only get the failure when you are specifically running docker registry "release maintenance" tasks; not as part of a prepareCompose.

I think it makes sense to add a "I know this is out of date, don't fail" flag. It will allow us to make sure that our failure notification to DHC notify is new information.

Reply:
Ah yes, compareImage, sorry. Will re-run with the latest and make sure I follow.

Reply:
Going through my bash history, I actually ran compareImage, so that's good at least... 

To confirm, you added the new ignoreOutOfDate on non-alpine images, rather than update them - is that what you meant to do? Without the last commit, wouldn't slack have started messaging nightly to fix these?

Also, maybe change the note when ignoreOutOfDate is set, to the effect of "this is out of date, ignoring, remove ignoreOutOfDate or run bumpImage and test" so it is clear which images are "really out of date" and which ones are "expected to be out of date"?

Or do we want to start from the default position of ignoreOutOfDate=true?

Reply:
I'm happy to bump all the digests to their latest repo digest version. The reason I didn't (yet - was going to be an immediate follow up to this PR) was to keep the build images as close as possible to the "existing" build images (at the time the PR first started).

The "out-of-date" refers to human-friendly name being out-of-date wrt the repo digest; it doesn't refer to the human-friendly name being out-of-date.  We should even bump alpine:3.12; there are likely security concerns sticking to old 3.12s.

Without my last commit, we would be getting notifications in the DHC notification channel nightly. With my latest commit, the relevant CI tasks would not fail, and would not cause notifications (but it would still print out warnings).

The default should be ignoreOutOfDate=false; if we are stuck on an explicit *repo-digest* (because the newer digest breaks our build, or because we are "baking" QA for a release), that is the time to set ignoreOutOfDate=false and likely attache a ticket number.

I'll push out what was going to be a follow-up PR to bump all the versions.